src/book/02-system/01-nubots/03-nuclear.mdx

```diff
@@ -5,3 +5,144 @@ title: NUClear
+| Single   | Is used to specify that only one instance of the associated reaction can execute at any given point during runtime. `on<Trigger<T, ...>, Single>()` will only allow one task for this reaction to be executed or in the queue at any given time. This should be fused with another DSL word, such as `Trigger` in the example.                                                                                                                                           |
+| Buffer   | Is used to specify that up to $n$ instances of the associated reaction can execute during runtime. `on<Trigger<T, ...>, Buffer<n>>>()` will create and schedule a new task if there are less than $n$ existing tasks for this reaction. This should be fused with another DSL word, such as `Trigger` in the example.                                                                                                                                   |
+| Priority | Assigns a priority to any tasks created to determine scheduling order in the PowerPlant. `on<Trigger<T, ...>, Priority::HIGH>()` will schedule the task with `HIGH` priority when the reaction is triggered. Available priority settings are `REALTIME`, `HIGH`, `NORMAL`, `LOW`, and `IDLE`. This should be fused with another DSL word, such as `Trigger` in the example.                                                                                      |
+| Sync     | Sets the synchronisation for a group of tasks. `on<Trigger<T, ...>, Sync<Group>>()` will allow only one task from `Group` to execute at any given time. This is similar to the `Single` word, but applies to multiple reactions. This word helps prevent race conditions. This should be fused with another DSL word, such as `Trigger` in the example.                                                                                   |
+| Every    | Used to request periodic reactions in the system. `on<Every<2, std::chrono::seconds>()` will create a task every two seconds. `on<Every<2, Per<std::chrono::seconds>>()` will create a task two times per second.                                                                                                                                                                                                                              |
+| Always   | Used to request continuous reactions in the system. `on<Always>` will ensure a single instance of the associated reaction is running at all times.                                                                                                                                                                                                                                                                                             |
```

Comment:
Maybe mention that on Always runs in its own thread rather than using the thread pool